app/src/main/java/unipd/se18/ocrcamera/SearchResultActivity.java

```diff
@@ -0,0 +1,407 @@
+package unipd.se18.ocrcamera;
+
+import android.app.SearchManager;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Process;
+import android.support.annotation.NonNull;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.Button;
+import android.widget.Filter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+import unipd.se18.ocrcamera.inci.IngredientsExtractor;
+
+
+/**
+ * This activity allow the user to manually search for ingredients, providing smart suggestions
+ * while the user is typing.
+ * Currently it is launched with an ACTION_SEARCH intent from the SearchView contained in the
+ * menu, inside ResultActivity.java
+ * @author Luca Moroldo g3
+ */
+public class SearchResultsActivity extends AppCompatActivity {
+
+    private static final String TAG = "SearchResultsActivity";
+
+    /**
+     * List used to show the results of a search
+     */
+    private ListView mIngredientsListView;
+
+    /**
+     * TexvView used to show messages when the result of a search is empty
+     */
+    private TextView mMessageTextView;
+
+    /**
+     * An extension of EditText that show suggestions with a dropdown list
+     */
+    private AutoCompleteTextView mAutoCompleteTextView;
+
+    /**
+     * Button that on click run a search and update mIngredientsListView with results
+     */
+    private Button mSearchButton;
+
+    /**
+     * Adapter used to update the dropdown suggestions list
+     */
+    private ArrayAdapter<String> dropDownListAdapter;
+
+    /**
+     * Used to find ingredients from the INCI db with a query
+     */
+    private IngredientsExtractor ingredientsExtractor;
+
+    /**
+     * Size of the dropdown suggestions list
+     */
+    private static final int MAX_DROPDOWN_SUGGESTIONS = 40;
+
+    /**
+     * Minimum query size to show suggestions
+     */
+    private static final int SUGGESTION_THRESHOLD = 2;
+
+
```

Comment:
the specifications are missing. Input, requirements, any outputs, descriptions, etc.

Reply:
This method it's a part of the activity lifecycle, you can check the documentation here: 
https://developer.android.com/guide/components/activities/activity-lifecycle#java
it's where all the components are usually initialized, anyway I'll add a comment about it, thank you for your report.